internal/runtimehooks/webhooks/server.go

```diff
@@ -105,9 +107,12 @@ func (s *Server) Start(ctx context.Context) error {
 		return err
 	}
 
+	httProxyConfig := httpproxyconfig.New()
 	allHandlers := []handlers.NamedHandler{
 		servicelbgc.New(client),
 		calico.New(client, *s.calicoCNIConfig),
+		name.NewDiscoveryVariables(name.Suffix("vars"), httProxyConfig),
+		name.NewGeneratePatches(name.Suffix("patch"), httProxyConfig),
```

Comment:
Do you really dislike the idea of doing this in the handler registration blocks rather than here?

Reply:
I've removed that part and split the handler into 2 separate handler:

```
http-proxy-vars
http-proxy-inject
```

Reply:
If we want to enforce a separate handler per hook type, we could set the function names in the interfaces to be the same but with different parameter types.

Reply:
I agree, but I think in this case it makes sense to split this handler into 2 separate handlers, as we could have multiple handlers that would somehow act on http proxy variable.